lsp/src/main/scala/org/ensime/lsp/rpc/companions.scala

```diff
@@ -0,0 +1,136 @@
+// Copyright: 2010 - 2017 https://github.com/ensime/ensime-server/graphs
+// License: http://www.gnu.org/licenses/gpl-3.0.en.html
+package org.ensime.lsp.rpc.companions
+
+import spray.json.{ JsObject, JsonFormat }
+import org.ensime.lsp.rpc.messages._
+import JsonRpcMessages._
+import org.ensime.lsp.rpc.JsInnerFormats.JsInnerField
+import shapeless.tag
+
+import scala.util.{ Failure, Success, Try }
+
+sealed trait RpcCompanionError {
+  val describe: String
```

Comment:
best to make this a sealed abstract class that takes a parameter. vals in traits are evil.